pennylane/transforms/qcut.py

```diff
@@ -1303,6 +1303,34 @@ def _wrapper(*args, **kwargs):
     return _wrapper
 
 
+def _qcut_expand_fn_mc(
+    tape: QuantumTape,
+    shots: Optional[int] = None,
+    device_wires: Optional[Wires] = None,
+    classical_processing_fn: Optional[callable] = None,
+    max_depth: int = 1,
+):
+    """Expansion function for sample-based circuit cutting.
+
+    Expands operations until reaching a depth that includes :class:`~.WireCut` operations.
+    """
+    # pylint: disable=unused-argument
+    for op in tape.operations:
+        if isinstance(op, WireCut):
+            return tape
+
+    if max_depth > 0:
+        return cut_circuit_mc.expand_fn(tape.expand(), max_depth=max_depth - 1)
+
+    raise ValueError(
+        "No WireCut operations found in the circuit. Consider increasing the max_depth value if "
+        "operations or nested tapes contain WireCut operations."
+    )
+
+
+cut_circuit_mc.expand_fn = _qcut_expand_fn_mc
+
+
```

Comment:
This works but I'm not sure if it makes sense to have `_qcut_expand_fn` and `_qcut_expand_fn_mc` be the same function with repeated code. How about:

```python
def _qcut_expand_fn(
    tape: QuantumTape,
    max_depth: int = 1,
):
    """Expansion function for sample-based circuit cutting.
    Expands operations until reaching a depth that includes :class:`~.WireCut` operations.
    """
    for op in tape.operations:
        if isinstance(op, WireCut):
            return tape

    if max_depth > 0:
        return _qcut_expand_fn(tape.expand(), max_depth=max_depth - 1)

    raise ValueError(
        "No WireCut operations found in the circuit. Consider increasing the max_depth value if "
        "operations or nested tapes contain WireCut operations."
    )


def cut_circuit_expand(
    tape: QuantumTape,
    use_opt_einsum: bool = False,
    device_wires: Optional[Wires] = None,
    max_depth: int = 1,
):
    # pylint: disable=unused-argument
    return _qcut_expand_fn(tape, max_depth)


def cut_circuit_mc_expand(
    tape: QuantumTape,
    shots: Optional[int] = None,
    device_wires: Optional[Wires] = None,
    classical_processing_fn: Optional[callable] = None,
    max_depth: int = 1,
):
    # pylint: disable=unused-argument
    return _qcut_expand_fn(tape, max_depth)


cut_circuit.expand_fn = cut_circuit_expand
cut_circuit_mc.expand_fn = cut_circuit_mc_expand
```
I'm not sure it'll be much shorter, but it seems cleaner for both expansions to be using the same core function (reducing code duplication).

That being said, the current approach works well if we expect `cut_circuit` and `cut_circuit_mc` expansions to ever behave differently :thinking: